src/client/api/json_parser/payloads.c

```diff
@@ -353,18 +353,28 @@ cJSON* json_tagged_serialize(tagged_data_t* tagged_data) {
 
   // tag
   if (tagged_data->tag) {
-    if (!cJSON_AddStringToObject(tagged_data_payload, JSON_KEY_TAG, (const char* const)tagged_data->tag->data)) {
-      printf("[%s:%d] add tag type failed\n", __func__, __LINE__);
+    char* tag_str = malloc(BIN_TO_HEX_STR_BYTES(tagged_data->tag->len));
+    if (!tag_str) {
+      printf("[%s:%d] OOM\n", __func__, __LINE__);
       cJSON_Delete(tagged_data_payload);
       return NULL;
     }
-  } else {
-    // add a null tag to tagged data
-    if (!cJSON_AddNullToObject(tagged_data_payload, JSON_KEY_TAG)) {
-      printf("[%s:%d] add null tag payload failed\n", __func__, __LINE__);
+
+    int bin_2_hex_ret =
+        bin_2_hex(tagged_data->tag->data, tagged_data->tag->len, tag_str, BIN_TO_HEX_STR_BYTES(tagged_data->tag->len));
+    if (bin_2_hex_ret != 0) {
```

Comment:
`bin_2_hex_ret` is not realy needed, the same as line [389](https://github.com/iotaledger/iota.c/pull/316/files#diff-f16bd3240fb36df6ddc7b455d53f2e34fbe0330fb3f54bd4a14d3cc1b8e00012R389)
```suggestion
    if (bin_2_hex(tagged_data->tag->data, tagged_data->tag->len, tag_str, BIN_TO_HEX_STR_BYTES(tagged_data->tag->len)) != 0) {
```

Reply:
Removed bin_2_hex_ret.